### PR TITLE
fix(AM-4461): fix goroutines leakage

### PR DESCRIPTION
**Pull request by rahulsawra98**
This PR fixes the exponentially increase of goroutines in gw sidecar.

### CODE REVIEW

File: pkg/status/tunnel.go [hunk 1]
```diff
@@ -124,23 +124,23 @@ func (t *TunnelChecker) Stop() error {
...
+		//do not start new pinger in case Peer IP is same
+		currentIp := fmt.Sprintf("%v", tunMsg.msg)
+		if t.tunStatus.PeerIP == currentIp {
+			return nil
 		}
-		err := t.startPing(t.tunStatus.PeerIP)
-		if err != nil {
-			t.log.Errorf("Ping Start failed: %s", err.Error())
-			return err
+		t.tunStatus.PeerIP = fmt.Sprintf("%v", tunMsg.msg)
```
Comment (rajendra-avesha) [inline pkg/status/tunnel.go:135]: I think we can use t.tunStatus.PeerIP = currentIp

**Review by rajendra-avesha — CHANGES_REQUESTED** [review]

File: go.mod [hunk 2]
```diff
@@ -50,6 +50,7 @@ require (
 	github.com/beorn7/perks v1.0.1 // indirect
 	github.com/cespare/xxhash/v2 v2.1.2 // indirect
 	github.com/matttproud/golang_protobuf_extensions v1.0.1 // indirect
+	github.com/pkg/profile v1.6.0 // indirect
```
Comment (rajendra-avesha) [inline go.mod:53]: Can we remove profile

File: main.go [hunk 3]
```diff
@@ -133,7 +133,6 @@ func main() {
 	go metrics.StartMetricsCollector(metricCollectorPort)
 
 	go shutdownHandler(wg)
-
```
Comment (rajendra-avesha) [inline main.go:136]: please remove this

**Review by rajendra-avesha — CHANGES_REQUESTED** [review]

File: pkg/status/tunnel.go (hunk 1 above)
Comment (rahulsawra98) [reply pkg/status/tunnel.go]: done

**Review by rajendra-avesha — APPROVED** [review]

**Pull request merged**